### ISSUE TITLE
External changes of open entry are not synced-back to the entry editor

### ISSUE THREAD

**Issue by tobiasdiez** [JabRef/jabref#2946]
Changes like generating a new bibtexkey or renaming the file are not displayed in the entry editor.

**Comment by lenhard** [comment]
@tobiasdiez Is it possible that the `ObjectBinding` between the editor and the `BibEntry` breaks when you change something in the map of its fields, for instance by calling `fields.put()`? Can you please have a look at that?

**Comment by lenhard** [comment]
@tobiasdiez Sorry, but this one is not fixed yet. It works now if you change something in the source tab. Then the changes are synced back to the field editor.

But it does not work in case of external changes. If you apply save actions, the changes only become visible if you reopen the file. If you change the file externally in a text editor, the changes become visible in the main table but not in the entry editor. They only become visible when the entry editor is recreated.

**Comment by tobiasdiez** [comment]
I tried it with the "convert to biblatex" cleanup and it worked (the year was moved to date and this change was correctly shown in the entry editor). Same with the "generate bibtex key" action.
What do I have to do to reproduce the problem you described?

**Comment by lenhard** [comment]
@tobiasdiez I'll have to correct myself regarding safe actions. They do work. But the external changes don't. Do the following:
 1. open a bib file in jabref and a text editor (e.g. notepad++)
 2. change the content of the author field in notepad to something else, say `test`
 3. Go back to JabRef, the files changed dialog pops up. Accept the changes
 4. Now, the new author is displayed in the main table, but not in the entry editor.

**Comment by tobiasdiez** [comment]
Ok, probably the merge dialog deletes the old bibentry and creates a new one instead of updating the one already present in the database. Thus the binding to the entry editor gets destroyed.
Should I create a new issue for this or do we silently ignore this problem until someone complains? 😄

**Comment by lenhard** [comment]
That's the clue. The review panel says that there's an entry added and an entry removed, so that's why the relation breaks.

So that's not really critical, since I don't see the use case. Why would someone change the exact same entry in JabRef and a text editor in parallel, except if it's me who's testing. If a different entry is changed, the entry editor is recreated anyway.

Let's keep this open and I'll rename it and remove it from the 4.0 milestone.

**Comment by lenhard** [comment]
The remaining part is fixed by #3156.
